app/waf/src/controllers/pool.controller.ts

```diff
@@ -29,16 +39,26 @@ export class PoolController {
     public memberRepository: MemberRepository,
   ) {}
 
+  readonly createDesc = 'Pool resource that need to be created';
   @post(prefix + '/pools', {
     responses: {
-      '200': {
-        description: 'Pool model instance',
-        content: {'application/json': {schema: {'x-ts-type': Pool}}},
-      },
+      '200': Schema.response(Pool, 'Successfully create Pool resource'),
+      '400': Schema.badRequest('Invalid Pool resource'),
+      '422': Schema.unprocessableEntity('Unprocessable Pool resource'),
     },
   })
-  async create(@requestBody() pool: Pool): Promise<Pool> {
-    return await this.poolRepository.create(pool);
+  async create(
+    @requestBody(Schema.createRequest(Pool, this.createDesc))
+    reqBody: Partial<Pool>,
+  ): Promise<PoolResponse> {
+    try {
+      reqBody.id = uuid();
```

Comment:
You may rebase the latest code in master branch. The uuid can be generated in repository layer now. And the we also have a new implementation of Response in controller/common.ts, so you needn't to create a PoolResponse class now.